include/unifex/tag_invoke.hpp

```diff
@@ -32,6 +32,21 @@ struct tag_invoke_cpo {
     return tag_invoke((CPO&&)cpo, (Args &&) args...);
   }
 };
+
+template<typename CPO, typename... Args>
+using tag_invoke_result_t = decltype(tag_invoke(static_cast<CPO&&(*)()>(nullptr)(), static_cast<Args&&(*)()>(nullptr)()...));
```

Comment:
Maybe use `noexcept` function pointers here as you do in `try_tag_invoke` below.